cocotb/decorators.py

```diff
@@ -327,35 +327,19 @@ class function(object):
     externally appear to yield.
     """
     def __init__(self, func):
-        self._func = func
+        self._raw_coro = func
```

Comment:
`func` was misleading, this has actually always been an object that contains `yield` / `await`.